controllers/openstackcluster_controller.go

```diff
@@ -249,7 +249,18 @@ func (r *OpenStackClusterReconciler) deleteBastion(ctx context.Context, scope *s
 		return err
 	}
 
+	var statusFloatingIP *string
+	var specFloatingIP *string
 	if openStackCluster.Status.Bastion != nil && openStackCluster.Status.Bastion.FloatingIP != "" {
+		statusFloatingIP = &openStackCluster.Status.Bastion.FloatingIP
+	}
+	if openStackCluster.Spec.Bastion.FloatingIP != nil {
```

Comment:
```suggestion
	if openStackCluster.Spec.Bastion != nil && openStackCluster.Spec.Bastion.FloatingIP {
```
Looks like this is the reason of failing e2e.